espnet/nets/pytorch_backend/e2e_asr_transformer.py

```diff
@@ -181,14 +193,48 @@ def forward(self, xs_pad, ilens, ys_pad):
 
         # TODO(karita) show predected text
         # TODO(karita) calculate these stats
-        cer, cer_ctc, wer = 0.0, 0.0, 0.0
-        if self.ctc is None:
+        if self.mtlalpha == 0.0:
             loss_ctc = None
+            cer_ctc = None
         else:
             batch_size = xs_pad.size(0)
             hs_len = hs_mask.view(batch_size, -1).sum(1)
             loss_ctc = self.ctc(hs_pad.view(batch_size, -1, self.adim), hs_len, ys_pad)
 
+            cers, char_ref_lens = [], []
```

Comment:
Can you make these entire lines (204 -- 237) as one function (class member function or function in some common directory)?
I want to make the main loop simple.